### PR TITLE
Enhanced tests - added negative checks

### PR DESCRIPTION
**Pull request by MuazOthman**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (ducaale) [inline .vscode/settings.json:1]: Can you revert unrelated file changes?

File: package-lock.json [hunk 2]
```diff
@@ -1,8 +1,5559 @@
 {
   "name": "@hu-bcs1/islamic-inheritance-calculator",
   "version": "1.4.2",
-  "lockfileVersion": 1,
+  "lockfileVersion": 2,
```
Comment (ducaale) [inline package-lock.json:4]: Same as above

File: __tests__/index.test.ts [hunk 3]
```diff
@@ -31,1073 +50,1197 @@ describe('Some edge cases', () => {
...
-test('1 daughter, 1 paternal_grand_daughter, 2 full_sister', () => {
-  const result = calculate({
-    daughter: 1,
-    paternal_grand_daughter: 1,
-    full_sister: 2
+  test('1 wife, 1 son', () => {
+    const result = calculate({ wife: 1, son: 1 })
+    checkResult(result, 'wife', f(1,8))
+    checkResult(result, 'son', f(7,8))
+    ensureCompleteResult(result, ['wife', 'son'])
```
Comment (ducaale) [inline __tests__/index.test.ts:59]: Should we have an assertion inside `calculate` that ensures there are no negative results? That way, we don't need this check at the end of each test.

File: __tests__/index.test.ts [hunk 4]
```diff
@@ -31,1073 +50,1197 @@ describe('Some edge cases', () => {
 })
 
 // following test cases were taken from http://inheritance.ilmsummit.org/projects/inheritance/testcasespage.aspx
+describe('ilmsummit test cases', () => {
```
Comment (ducaale) [inline __tests__/index.test.ts:53]: Until we add tests from different sources, can we remove this describe block?

File: __tests__/index.test.ts [hunk 5]
```diff
@@ -31,1073 +50,1197 @@ describe('Some edge cases', () => {
...
-  const result = calculate({
-    husband: 1,
-    mother: 1,
-    maternal_sibling: 1,
-    full_paternal_uncle: 1
-  })
-  checkResult(result, 'husband', f(1,2))
-  checkResult(result, 'mother', f(1,3))
-  checkResult(result, 'maternal_sibling', f(1,6))
-  checkResult(result, 'full_paternal_uncle', f(0))
```
Comment (ducaale) [inline __tests__/index.test.ts:51]: Can you add back any `checkResult` that has been removed?

File: __tests__/index.test.ts [hunk 6]
```diff
@@ -31,1073 +50,1197 @@ describe('Some edge cases', () => {
...
+    checkResult(result, 'mother', f(1,3))
+    checkResult(result, 'paternal_grand_father', f(1,6))
+    ensureCompleteResult(result, ['husband', 'mother', 'paternal_grand_father'])
+  })
 
-test('husband, mother, 2 paternal_brother, 2 maternal_brother', () => {
-  const result = calculate({ husband: 1, mother: 1, paternal_brother: 2, maternal_sibling: 2 })
-  checkResult(result, 'husband', f(1,2))
-  checkResult(result, 'mother', f(1,6))
-  checkResult(result, 'paternal_brother', f(0,6))
```
Comment (ducaale) [inline __tests__/index.test.ts:1042]: Same as above.

There might be other instances I missed since this review was done from a phone 🙂

**Review by ducaale — CHANGES_REQUESTED** [review]
Thanks!

Do you mind adding a brief description to the PR's body?
